app/models/alchemy/page/page_elements.rb

```diff
@@ -134,6 +134,16 @@ def available_element_names
       @_available_element_names ||= available_element_definitions.map { |e| e['name'] }
     end
 
+    # Available element definitions excluding nested unique elements.
+    #
+    def available_elements_within_current_scope(parent)
+      @_available_elements = if parent
+        available_element_definitions(parent.try(:name)).reject{ |e| parent.nested_elements.select(&:unique).map(&:name).include? e["name"] }
```

Comment:
As you already know that you have a `parent` (through the `if` one line above), you can omit the `try` and can directly use `parent.name`